pkg/collector/corechecks/oracle/activity.go

```diff
@@ -0,0 +1,211 @@
+		blocking_session
+	ELSE
+		null
+	END blocking_session,
+	CASE WHEN final_blocking_session_status = 'VALID' THEN
+    	final_blocking_instance
+	ELSE
+		null
+	END final_blocking_instance,
+	CASE WHEN final_blocking_session_status = 'VALID' THEN
+    	final_blocking_session
+	ELSE
+		null
+	END final_blocking_session,
+    CASE WHEN state = 'WAITING' THEN
+		event
+	ELSE
+		'CPU'
+	END event,
+	CASE WHEN state = 'WAITING' THEN
+    	wait_class
+	ELSE
+		'CPU'
+	END wait_class,
+	sql_text,
+	pdb_name
+FROM sys.dd_session
+WHERE 
+	( sql_text NOT LIKE '%DD_ACTIVITY_SAMPLING%' OR sql_text is NULL ) 
+	AND (
+		NOT (state = 'WAITING' AND wait_class = 'Idle')
+		OR state = 'WAITING' AND event = 'fbar timer' AND type = 'USER'
+	)`
+
+type RowMetadata struct {
+	Commands       []string `json:"dd_commands,omitempty"`
+	Tables         []string `json:"dd_tables,omitempty"`
+	Comments       []string `json:"dd_comments,omitempty"`
+	QueryTruncated string   `json:"query_truncated,omitempty"`
+}
+
+type OracleActivityRow struct {
+	SessionID              uint64  `db:"SID" json:"sid,omitempty"`
+	SessionSerial          uint64  `db:"SERIAL#" json:"serial,omitempty"`
+	Username               *string `db:"USERNAME" json:"username,omitempty"`
+	OsUser                 *string `db:"OSUSER" json:"os_user,omitempty"`
+	Process                *string `db:"PROCESS" json:"process,omitempty"`
+	Machine                *string `db:"MACHINE" json:"machine,omitempty"`
+	Program                *string `db:"PROGRAM" json:"program,omitempty"`
+	Type                   *string `db:"TYPE" json:"type,omitempty"`
+	SqlID                  *string `db:"SQL_ID" json:"sql_id,omitempty"`
+	ForceMatchingSignature *uint64 `db:"FORCE_MATCHING_SIGNATURE" json:"force_matching_signature,omitempty"`
+	SqlPlanHashValue       *uint64 `db:"SQL_PLAN_HASH_VALUE" json:"sql_plan_hash_value,omitempty"`
+	SqlExecStart           *string `db:"SQL_EXEC_START" json:"sql_exec_start,omitempty"`
+	Module                 *string `db:"MODULE" json:"module,omitempty"`
+	Action                 *string `db:"ACTION" json:"action,omitempty"`
+	ClientInfo             *string `db:"CLIENT_INFO" json:"client_info,omitempty"`
+	LogonTime              *string `db:"LOGON_TIME" json:"logon_time,omitempty"`
+	ClientIdentifier       *string `db:"CLIENT_IDENTIFIER" json:"client_identifier,omitempty"`
+	BlockingInstance       *uint64 `db:"BLOCKING_INSTANCE" json:"blocking_instance,omitempty"`
+	BlockingSession        *uint64 `db:"BLOCKING_SESSION" json:"blocking_session,omitempty"`
+	FinalBlockingInstance  *uint64 `db:"FINAL_BLOCKING_INSTANCE" json:"final_blocking_instance,omitempty"`
+	FinalBlockingSession   *uint64 `db:"FINAL_BLOCKING_SESSION" json:"final_blocking_session,omitempty"`
+	Event                  *string `db:"EVENT" json:"event,omitempty"`
+	WaitClass              *string `db:"WAIT_CLASS" json:"wait_class,omitempty"`
+	SqlText                *string `db:"SQL_TEXT" json:"sql_text,omitempty"`
```

Comment:
I'm not sure the agent linting checks this but we'll have to uppercase `SQL` and it's going to end up `SQLText` when we get in dd-go.

Reply:
Renamed `SqlText` to `Statement`. Will name acronyms in capital letter from now on.

Reply:
That works!